books/views.py

```diff
@@ -56,41 +75,56 @@ def read_or_update(request, book_id):
+            serializer = AuthorSerializer(author, context={'request': request})
+            return Response(serializer.data, status=status.HTTP_200_OK)
 
     def post(self, request):
-        author = request.data.get('author')
-        serializer = AuthorSerializer(data=author)
+        serializer = AuthorSerializer(data=request.data, context={'request': request})
         if serializer.is_valid():
-            try:
-                serializer.save()
-            except IntegrityError:
-                return Response({"message": "Integrity Error"}, status=status.HTTP_400_BAD_REQUEST)
-            return Response({
-                "message": "successfully created author",
-                "data": serializer.data
-            }, status=status.HTTP_201_CREATED)
+            if request.user == serializer.validated_data.get('user'):
+                try:
+                    serializer.save()
+                except IntegrityError:
+                    return Response({
+                        "message": "Integrity Error"
+                    }, status=status.HTTP_400_BAD_REQUEST)
+                return Response({
+                    "message": "successfully created author",
+                    "data": serializer.data
+                }, status=status.HTTP_201_CREATED)
+            else:
+                return Response({
+                    "message": "You can not create author for another user account other than your own"
+                }, status=status.HTTP_400_BAD_REQUEST)
         else:
             return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
 
-
-class AuthorDetail(APIView):
-    def get(self, request, pk, format=None):
-        author = get_object_or_404(Author, pk=pk)
-        serializer = AuthorSerializer(author)
-        return Response(serializer.data)
-
     def put(self, request, pk):
         author = get_object_or_404(Author, pk=pk)
-        data = request.data.get('author')
-        serializer = AuthorSerializer(instance=author, data=data, partial=True)
-        if serializer.is_valid(raise_exception=True):
-            serializer.save()
-        return Response(serializer.data)
+        if request.user == author.user:
+            data = request.data
+            serializer = AuthorSerializer(instance=author, data=data, partial=True, context={'request': request})
+            if serializer.is_valid(raise_exception=True):
+                serializer.save()
+            return Response(serializer.data, status=status.HTTP_200_OK)
+        else:
+            return Response({"message": "You cannot update an object other than your own"},
+                            status=status.HTTP_400_BAD_REQUEST)
 
     def delete(self, request, pk, format=None):
-        get_object_or_404(Author, pk=pk).delete()
-        return Response(status=status.HTTP_204_NO_CONTENT)
+        author = get_object_or_404(Author, pk=pk)
+        if request.user == author.user:
```

Comment:
Try to handle this through custom Django permission classes. 

Django view should be clean. All validations should be handled in serializes and permissions should be handled through custom permission classes.

Reply:
sure

Reply:
done